### ISSUE TITLE
critical damage calculation

### ISSUE THREAD

**Issue by beauxq** [EdenServer/community#47]
(This is a bug that Icon fixed on Kupo in September 2017.)

Right now on Eden, the cap for critical damage is the same as the cap for normal damage.

For crits, the cap for the multiplier should be 3.

You can see the 3 cap hasn't changed over the history:
http://ffxiclopedia.wikia.com/wiki/Level_Correction_Function_and_pDIF
http://ffxiclopedia.wikia.com/wiki/Level_Correction_Function_and_pDIF?oldid=85991

Here is a fix:
src/map/utils/battleutils.cpp

```
    if (PAttacker->objtype == TYPE_PC)
        {
---         ratioCap = 2.25f;
+++         ratioCap = isCritical ? 3 : 2.25f;
        }
```

**Comment by beauxq** [comment]
https://github.com/EdenServer/community/issues/45

another issue made at about the same time

**Comment by beauxq** [comment]
@godmode 
This is not fixed yet on Eden.
I just tested the code diff on my DSP again and it fixes it there.

Want to check on it again? or have you changed something else in that function that would affect it?

**Comment by godmode** [comment]
You’re right. The server got rebooted; never went down. So, yeah. Will be next Sunday when it gets compiled again.

**Comment by FallenXI** [comment]
This is still not fixed.

**Comment by godmode** [comment]
Can you provide more data?

**Comment by FallenXI** [comment]
I've only got this one screenshot at the moment, as my chat log just screwed up trying to screenshot it and deleted, but:

<img width="417" alt="Screenshot 2019-03-13 at 15 56 23" src="https://user-images.githubusercontent.com/48524641/54293969-c6953400-45a8-11e9-89d0-60bb44f757eb.png">

I've never seen an instance where crits are this close to normal damage. I tested this crit issue on another server and my data can be found here (we didn't have the code from Kupo, sadly):

https://docs.google.com/spreadsheets/d/12tTt_oOEgMeDbq9R8FnRKRtq7Tclom3hEJDvz7mU7zs/edit?usp=sharing

The first sheet can mostly be ignored as I'm not sure I calculated the qRatio etc correctly, but the raw crit/normal testing data and the averages at the bottom show the difference in crits vs retail. 

It basically boils down to:

<img width="608" alt="Screenshot 2019-03-13 at 16 04 06" src="https://user-images.githubusercontent.com/48524641/54294770-ba5da680-45a9-11e9-968d-9b6f34678f70.png">

Granted, the left column in the pic is a different server, but already there is more correlation between that server and Eden, than retail.

**Comment by beauxq** [comment]
The difference (percentage difference) between crits and normal hits is not the same in all situations. The data you've posted for retail is not the same for all situations on retail. In some situations on retail, crits do get very close to normal hits, in other retail situations, crits don't get that close to normal hits.

So the fact that the "normal" numbers don't match up between the data you've shown, shows that they are not in the same situation. So that would make that data irrelevant even if it were from Eden.

I don't think Eden is perfect, but it's a lot closer to retail than it is to DSP.

**Comment by FallenXI** [comment]
If you can show me proof that retail has crit numbers this close to normal
attacks, then consider this closed - but they still look to me like they
are not working as they should.

**Comment by beauxq** [comment]
https://vignette.wikia.nocookie.net/ffxi/images/4/45/PDIF.gif/revision/latest?cb=20080111144558

Here you can see the era pdif graph with upper and lower boundaries for normal hits and for crits.

With Atk/Def Ratio of 2, you can see normal hits can go up to a multiplier of 2.4, and crits can go as low as 2.6

So if the damage calculation before the multiplier is something like 6.8, the normal damage can go up to 16.32, and the crits go as low as 17.68
That fits what's in the screenshot above.

Notice that that is not the case for all Atk/Def Ratios.

If the Atk/Def Ratio is 1.25, then the highest normal is 1.5 and the lowest crit is 2.
So if the damage calculation before the multiplier is 10, that makes the highest normal 15, and the lowest crit is 20. So that would not allow for what's in the screenshot. But I doubt the Atk/Def Ratio of someone getting 16 damage on an Orcish Mesmerizer is that low.

**Comment by FallenXI** [comment]
Thank you for the graph and explanation - I wouldn't worry about this issue
further, as most people I've spoken to since posting this say it seems ok
to them.

Sorry for the necrobump!
